projects/igniteui-angular/src/lib/directives/forms-control/forms-control.directive.ts

```diff
@@ -0,0 +1,99 @@
+import {
+    Directive,
+    forwardRef,
+    ElementRef,
+    HostListener,
+    NgModule,
+    Renderer2,
+    AfterViewInit,
+    OnDestroy
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Directive({
+    selector: 'igc-rating', /* eslint-disable-line @angular-eslint/directive-selector */
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => IgxFormsControlDirective),
+            multi: true
+        }
+    ]
+})
+export class IgxFormsControlDirective implements ControlValueAccessor, AfterViewInit, OnDestroy {
+    /** @hidden @internal */
+    public onChange: any = () => { };
+    /** @hidden @internal */
+    public onTouched: any = () => { };
+
+    protected _destroy$ = new Subject();
+
+    private _value: any;
+
+    /** @hidden @internal */
+    public get value(): any {
+        return this.elementRef.nativeElement.value;
+    }
+    public set value(val) {
+        if (val !== this._value) {
+            this._value = val;
+            this.onChange(this._value);
+            this.elementRef.nativeElement.value = val;
+        }
+    }
+
+    constructor(
+        private elementRef: ElementRef,
+        private renderer: Renderer2) { }
+
+    /** @hidden @internal */
+    public ngAfterViewInit(): void {
+        fromEvent(this.elementRef.nativeElement, 'blur')
```

Comment:
So the change is hooked up with `HostListener` and this through rxjs? Thinking you can get away doing both as host listeners and drop the destroy & init hooks and the destroyed subject